mc2/controllers/base/views.py

```diff
@@ -192,3 +193,27 @@ def post(self, request, controller_pk):
             messages.error(self.request, msg)
             return HttpResponseBadRequest(msg)
         return redirect('home')
+
+
+class ControllerWebhookRestartView(View):
+    """
+    Unauthenticated webhook to restart an app.
+
+    Technically, this is a 'URL capability'.
+    """
+
+    def get(self, request, controller_pk, token):
+        return HttpResponseNotAllowed(['POST'])
+
+    def post(self, request, controller_pk, token):
+        controller = get_object_or_404(Controller, pk=controller_pk)
+        if str(controller.webhook_token) != token:
+            return HttpResponseNotFound()
```

Comment:
`403 Forbidden` means "this thing exists but you're not authorised to have it", `404 Not Found` means "this thing doesn't exist". The client is making a request to `http://example.com/foo/bar/token/` without auth, so `404` is the more appropriate response if the token doesn't match.